app/service/auth.py

```diff
@@ -0,0 +1,62 @@
+import bcrypt, jwt
+
+from sqlmodel import Session, select
+from fastapi import FastAPI, status
+from fastapi.exceptions import HTTPException
+from typing import Optional
+from jose import JWTError
+
+from models.models import User, LoginRequest, TokenData
+from db import engine, SECRET_KEY, ALGORITHM
+
+app = FastAPI
+
+def get_access_token_after_succesful_login(login_request: LoginRequest) -> dict:
+    user = _authenticate_user(email=login_request.email, password=login_request.password)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect email or password",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+   
+    access_token = _create_access_token(email=user.email)
+    return {"access_token": access_token, "token_type": "bearer"}
+
+def get_current_user(token:str) -> Optional[User]:
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="could not validate credentials"
+    )
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=ALGORITHM)
+        email: str = payload.get("email")
+        
+        if email is None:
+            raise credentials_exception
+        token_data = TokenData(email = email)
+
+        with Session(engine) as session:
+            user = session.exec(select(User).filter(User.email == token_data.email)).first()
```

Comment:
```suggestion
        if email is None:
            raise credentials_exception

        with Session(engine) as session:
            user = session.exec(select(User).filter(User.email == email)).first()
```
P2: TokenData가 불필요한 것 같습니다. 어차피 email 필드만 담는 객체를 생성해서 token_data에 할당하고, 바로 밑에서 token_data.email로 이메일 정보를 가져오고 있어서 굳이 TokenData 클래스에 담아야하는 이유가 없는 것 같습니다.

Reply:
네! 수정하였습니다!